src/gui/dialogs/editor/edit_pbl.cpp

```diff
@@ -394,13 +395,22 @@ void editor_edit_pbl::select_icon_file()
 	if(dlg.show()) {
 		std::string path = dlg.path();
 		if(path.find(filesystem::get_core_images_dir()+"/icons/") == 0) {
-			std::string icon = path.substr(filesystem::get_core_images_dir().length());
+			std::string icon = path.substr(filesystem::get_core_images_dir().length()+1);
 			// setting this programmatically doesn't seem to trigger connect_signal_notify_modified()
 			find_widget<text_box>(get_window(), "icon", false).set_value(icon);
 			find_widget<drawing>(get_window(), "preview", false).set_label(icon);
 		} else {
-			// TODO: convert image file to data uri
-			// filesystem::scoped_istream nonmainline_icon = filesystem::istream_file(path);
+			// TODO: needing two vectors here is dumb
+			std::vector<char> t = filesystem::read_file_binary(path);
```

Comment:
Can't you just make it return unsigned to begin with?

Reply:
Yeah, turns out that does work - I was assuming it wouldn't be able to handle that.